### DISCUSSION TITLE
Editor: Spreadsheet format/Table improvements

### DISCUSSION THREAD

**Discussion by gsantner** [gsantner/markor#1667] [Ideas] [answered]
What I generally miss on mobile is a somewhat usable solution for spreadsheets & table editing - even more, that works with plaintext and no binary/proprietary format. Tables on mobile generally are a bit troublesome due limited width and unless you make tables with one or two word on 2 columns. You are out of horizontal space quickly, i.e. you can only see the first few column(s) with the others are in next line (automatic line break), or invisible out of screen (no automatic line break).

## Ideas
* Implement support for an plaintext format specalized for spreadsheet/tables (i.e. no general purpose format for texts like markdown/Zim) --> CSV?
* Way better support for table editing in Markdown format

Breaking one row **at editor** to multiple lines(/=rows) again to fit on screen generally defeats the purpose of tables/spreadsheet. Screen width also don't get magically larger. So the options are limited to:

* no automatic line break / horizontal scrolling / remaining columns invisible
* controlling content size / font size / scale

## Existing topics
Issues/Discussions with existing comments/discussion: 

* [Markdown Table Editor #338](https://github.com/gsantner/markor/issues/338)
* [Difficulty with Tables #1574](https://github.com/gsantner/markor/issues/1574)
* [md: Break line inside tables (two spaces at end of line) #1567 ](https://github.com/gsantner/markor/issues/1567)
* [markdown: auto format tables #1551](https://github.com/gsantner/markor/issues/1551)
* [md: auto format table rows (auto add new on enter press) #1058](https://github.com/gsantner/markor/issues/1058)
* [Markdown table and image size rendering #312](https://github.com/gsantner/markor/issues/312)

## Discussion
Discussion is very welcome, as well helping with development on topic :smiley:

**Comment by k3b** [comment, marked as answer]
Related: Feature request: https://github.com/gsantner/markor/issues/1980
Implemented in https://github.com/gsantner/markor/pull/1987 and https://github.com/gsantner/markor/pull/1988
